src/AttendanceLogger.java

```diff
@@ -0,0 +1,95 @@
+package src;
+
+import java.util.*;
+
+public class AttendanceLogger {
+    private HashMap<String, ArrayList<String>> attendance;
```

Comment:
Необходимо работать от интерфейсов, иначе это разрушает абстракцию и делает невозможным подстановку типа.
```suggestion
    private Map<String, List<String>> attendance;
```